samza-core/src/main/java/org/apache/samza/zk/BarrierForVersionUpgrade.java

```diff
@@ -19,29 +19,23 @@
 
 package org.apache.samza.zk;
 
-import java.util.List;
-
-
 /**
  * Interface for a barrier - to allow synchronization between different processors to switch to a newly published
  * JobModel.
  */
 public interface BarrierForVersionUpgrade {
   /**
    * Barrier is usually started by the leader.
-   * @param version - for which the barrier is started.
-   * @param processorsNames - list of processors available at the time of the JobModel generation.
    */
-  void start(String version, List<String> processorsNames);
+  void start();
```

Comment:
start() seems to be doing a lot more than starting a service. It is creating the barrier related subtree(s), setting a timer etc. It seems more apt to call it "prepareBarrier" with version and list of processors (as you had before). 
Otherwise, this information (about version and list of processors) is passed through the constructor of the implementation. Since it is not meant to be generic, why keep the interface vague?